src/coins.h

```diff
@@ -166,6 +166,7 @@ struct CCoinsCacheEntry
     {
         Assume(flags & (DIRTY | FRESH));
         if (!self.second.m_flags) {
+            Assume(self.second.m_prev == nullptr && self.second.m_next == nullptr);
```

Comment:
In commit "coins, refactor: Assume state after SetClean in AddFlags to prevent dangling pointers" (2d691b50d3729445a60898981b78c4239f2f0dd7)

I think this could also check the converse, that pointers are valid if flags were set. It could simplify check that pointers are non-null if flags are set, or go further and check prev->next == self, and next->prev == self

Reply:
> check that pointers are non-null if flags are set, or go further and check prev->next == self, and next->prev == self

the latter seems redundant, but the first is simple to check - done